WalletWasabi/Services/WalletService.cs

```diff
@@ -632,17 +633,11 @@ private async Task<bool> ProcessTransactionAsync(SmartTransaction tx)
 					walletRelevant = true;
 
 					foundKey.SetKeyState(KeyState.Used, KeyManager);
-					List<SmartCoin> spentOwnCoins = Coins.Where(x => tx.Transaction.Inputs.Any(y => y.PrevOut.Hash == x.TransactionId && y.PrevOut.N == x.Index)).ToList();
+					spentOwnCoins = spentOwnCoins ?? Coins.Where(x => tx.Transaction.Inputs.Any(y => y.PrevOut.Hash == x.TransactionId && y.PrevOut.N == x.Index)).ToList();
```

Comment:
No need to re-build the list of spent coins for each output received because the spent coin list doesn't change.